src/diagnostics.ts

```diff
@@ -32,7 +33,8 @@ export class DiagnosticsPublisher implements DiagnosticsHandler {
 	 * not present in subsequent updates.
 	 * @param diagnostics
 	 */
-	updateFileDiagnostics(diagnostics: ts.Diagnostic[]): void {
+	updateFileDiagnostics(diagnostics: ts.Diagnostic[], childOf = new Span()): void {
+		const span = childOf.tracer().startSpan('Publish diagnostics', { childOf });
 
 		// categorize diagnostics by file
 		const diagnosticsByFile = this.groupByFile(diagnostics);
```

Comment:
the operation needs to be wrapped in a try/catch/finally because spans must always be finished. You can find many examples of that in the project, it's always the same pattern